examples/rbd/storageclass.yaml

```diff
@@ -29,6 +29,8 @@ parameters:
    # eg: pool: rbdpool
    pool: <rbd-pool-name>
 
+   # Set thickprovision to true if you want thick rbd image.
+   thickprovision: "true"
```

Comment:
```suggestion
   thickProvision: "true"
```

Reply:
@dillaman I still need to add E2E testing for this new feature. this PR is created to measure the performance

Reply:
Will be happy to see the performance. Do we know what is the optimal block size to pass to Ceph?

Reply:
@mykaul  it's currently using default values we are not passing block-size parameter here. @dillaman any suggestion on settting block size? do we need to make it as a parameter is storageclass? or in cephcsi, we can calculate the required block size?

Reply:
@dillaman would writing the size of the stripe-unit be efficient? Or is there a more appropriate size that can be used with `rbd_writesame()`?

Reply:
Actually, I think using the stripe-unit would be the most inefficient. We use the stripe period in the `rbd` CLI when thick provisioning, which is just the `object-size` multiplied by the `stripe-count`.

Reply:
oh, wow :roll_eyes: thanks for the clarification!

Reply:
We should probably clarify that I wouldn't realistically expect much (if any) performance difference between thick and thin-provisioned images. The rationale is that bluestore will double-write on small IOs (< 4KiB) and it will perform redirect-on-write for all larger IOs. The benefit for thick-provisioning would be for better space accounting in your cluster when (potentially) overprovisioning.

Reply:
_better space accounting_ is what the most important change for users is expected to be. We have had reports where overprovisioning caused issues when the cluster ran out of space.

The updated implementation uses go-ceph `WriteSame()` with a `StripeUnit * StripeCount` block-size of zeros. If that is not correct, please let me know.

Thanks!

Reply:
@nixpanic Substitute `StripeCount` for `ObjectSize` (defaults to 4MiB / 1 << `Order`)

Reply:
@dillaman is there a rbd API that gets the ObjectSize from a created image? Or is it part of the pool? Just wondering how to get the value in case it has been reconfigured somewhere else...

Reply:
@nixpanic Yup, it's in `rbd_image_info_t` exposed via `rbd_stat`

Reply:
Thanks, that helps. 

> We use the stripe period in the rbd CLI when thick provisioning, which is just the object-size multiplied by the stripe-count.

:point_up: is what I implemented now too.